walk-in-interview/src/main/webapp/script.js

```diff
@@ -187,33 +212,104 @@ async function renderJobListings(sortBy, order, pageSize, pageIndex) {
     jobTitle.innerText = job['jobTitle'];
 
     const jobAddress = jobListing.children[1];
-
     const location = job['jobLocation'];
     jobAddress.innerText = `${location['address']}, ${location['postalCode']}`;
 
+    const jobPay = jobListing.children[2];
+    const pay = job['jobPay'];
+    jobPay.innerText = `${pay['min']} - ${pay['max']} SGD ` +
+      `(${pay['paymentFrequency'].toLowerCase()})`;
+
+    const requirementsList = jobListing.children[3];
+    const fullRequirementsList = getRequirementsList();
+    const requirementsArr = [];
+
+    job['requirements'].forEach((req) => {
+      requirementsArr.push(fullRequirementsList[req]);
+    });
+
+    requirementsList.innerText =
+      `Requirements List: ${requirementsArr.join(', ')}`;
+
     jobListingsElement.appendChild(jobListing);
   });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Add the list of jobs that are stored in the database
+ * given the filter fields.
+ */
+async function renderJobListings() {
+  if (!validateFilters()) {
+    return;
+  }
+
+  const sortingParam = document.getElementById('sort-by').value;
+
+  /** Note that this platform currently only sorts by Salary.*/
+  if (!sortingParam.includes(SALARY_PARAM)) {
+    console.log('error', 'this app currently only sorts by salary');
+    return;
+  }
+
+  const sortByParam = SALARY_PARAM;
+  const orderByParam = sortingParam.substring(7);
+
+  const regionParam = document.getElementById('show-by-region').value;
+  let minLimitParam = document.getElementById('filter-min-limit')
+      .valueAsNumber;
+  let maxLimitParam = document.getElementById('filter-max-limit')
+      .valueAsNumber;
+
+  if (Number.isNaN(minLimitParam)) {
+    minLimitParam = 0;
+  }
+
+  if (Number.isNaN(maxLimitParam)) {
+    maxLimitParam = JAVA_INTEGER_MAX_VALUE;
+  }
+
+  const jobPageData = await getJobListings(regionParam, sortByParam,
+      minLimitParam, maxLimitParam, orderByParam,
+      DEFAULT_PAGE_SIZE, DEFAULT_PAGE_INDEX)
+      .catch((error) => {
+        console.log('error', error);
```

Comment:
nit, for javascript console log i recommend you put some more context since you don't have stacktraces:

`console.log('Error fetching job listings', error);`